edb/repl/utils.py

```diff
@@ -127,16 +127,21 @@ def normalize_name(name: str) -> str:
 
 
 def get_filter_based_on_pattern(pattern: str,
-                                field: str = '.name',
+                                fields: Sequence[str] = ('.name',),
                                 flag: str = '') -> Tuple[str, Dict[str, str]]:
     if flag:
         flag = f'(?{flag})'
 
     qkw = {}
 
     if pattern:
-        qkw[f're_{field}'] = flag + pattern
-        return f'FILTER re_test(<str>$`re_{field}`, {field})', qkw
+        # the pattern is the same for all fields, so we only use one
+        # variable
+        qkw[f're_{fields[0]}'] = flag + pattern
```

Comment:
Please replace `f're_{fields[0]}'` with `'re_filter'`. This business with using the first element of the `fields` mapping for a parameter name is extremely confusing.

Reply:
Yeah, at this point that's probably better. Originally I just thought that it made sense to correlate the name of the regexp parameter with the name of the prop it's applied to, but with this update it's no longer the case.